### ISSUE TITLE
WPF Input Issues

### ISSUE THREAD

**Issue by ocardoso** [cefsharp/CefSharp#3300]
Hello!
 
- **What version of the product are you using?**
    86.0.241.0
 
- **What architecture x86 or x64?**
x64
   
- **On what operating system?**
Win10 Home, but seems isolated to something very narrow. We can reproduce always on HP Envy TouchScreen x360. Other customers have reported it.. I'm assuming on a variety of machines.
 
- **Are you using `WinForms`, `WPF` or `OffScreen`?**
WPF
   
- **What steps will reproduce the problem?**
- **What is the expected output? What do you see instead?**
 
Just run CefSharp.MinimalExample.Wpf
 
- Google should load up, and I should be able to type in the search box.
- The search box does not show up until you resize
- Text entered does not show up unless you resize
 
This does not happen with WinForms or CefClient. I recorded all this in action:

https://intuit.box.com/s/p0om1cg8imnq0gl4rzfksuwmzutdii05
 
Note, this is a brand new machine from BestBuy - I purchased it and reproduced, because a customer reported this on both the two HP Envy TouchScreen Machines he had purchased.  I have not been able to figure out what is special about this brand - if it's a setting, something that was installed by default, etc.  Note we are seeing this in our product in a few cases - i doubt specific to this brand - it's just the first time we've been able to reproduce and therefore show our code isn't involved (since we can reproduce with the Wpf Sample App)
 
 
- **Please provide any additional information below.*
 
    - Does the cef log provide any relevant information? (By default there should be a debug.log file in your bin directory)
 
Looks promising:

[1128/211214.266:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1128/211215.396:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1128/211215.976:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1128/211217.150:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1128/211223.068:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/?gws_rd=ssl (0)
[1128/211316.385:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1128/211317.185:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1128/211317.330:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1128/211317.356:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1128/211319.026:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1128/211343.940:ERROR:device_event_log_impl.cc(208)] [21:13:43.939] Bluetooth: bluetooth_adapter_winrt.cc:716 GetBluetoothAdapterStaticsActivationFactory failed: Class not registered (0x80040154)
[1129/071854.706:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1129/071854.957:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1129/071855.405:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1129/071856.320:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/071858.132:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/?gws_rd=ssl (0)
[1129/071917.697:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1129/072010.133:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1129/072010.223:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1129/072010.407:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1129/072011.423:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/072012.622:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1129/072210.265:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1129/072305.919:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/072306.539:INFO:CONSOLE(1)] "Unrecognized feature: 'vr'.", source: https://s.yimg.com/rq/darla/4-6-0/js/g-r-min.js (1)
[1129/072306.594:INFO:CONSOLE(1)] "Uncaught ReferenceError: Notification is not defined", source: https://s.yimg.com/aaq/notifications/js/sh-5.6.8.js (1)
[1129/072306.671:INFO:CONSOLE(3)] "CMP: Received error response retrieving GDPR consent from GUCE", source: https://s.yimg.com/aaq/cmp/version/3.0.3/cmp.js (3)
[1129/072306.696:ERROR:batching_media_log.cc(38)] MediaEvent: {"error":"FFmpegDemuxer: no supported streams"}
[1129/072306.702:INFO:CONSOLE(1)] "[CRAFT:Anchor "undefined" not present in current window context]", source: https://cdn.cmp.advertising.com/libraries/Delivr-2.7.4.js (1)
[1129/072306.712:ERROR:batching_media_log.cc(35)] MediaEvent: {"pipeline_error":14}
[1129/072306.713:INFO:CONSOLE(21)] "darla csc writer, potential invalid html content", source: https://s.yimg.com/rq/darla/4-6-0/html/r-csc.html (21)
[1129/072306.732:INFO:CONSOLE(1)] "Uncaught ReferenceError: Notification is not defined", source: https://s.yimg.com/aaq/notifications/js/sh-5.6.8.js (1)
[1129/072306.838:INFO:CONSOLE(308)] "[ybar debug] Nav-hide init triggered immediately", source: https://www.yahoo.com/ (308)
[1129/072306.853:INFO:CONSOLE(458)] "[ybar mailcount] nav calling unseen", source: https://www.yahoo.com/ (458)
[1129/072306.917:INFO:CONSOLE(1)] "TypeError: Cannot read property 'getSubscription' of undefined", source: https://www.yahoo.com/service-worker.js (1)
[1129/072307.175:INFO:CONSOLE(3)] "The provided value 'undefined' is not a valid enum value of type XMLHttpRequestResponseType.", source: https://yep.video.yahoo.com/oath/js/1/oath-player.js?expb=FPDOATH204&ypv=8.2.104 (3)
[1129/072307.189:INFO:CONSOLE(3)] "The provided value 'undefined' is not a valid enum value of type XMLHttpRequestResponseType.", source: https://yep.video.yahoo.com/oath/js/1/oath-player.js?expb=FPDOATH204&ypv=8.2.104 (3)
[1129/072307.333:INFO:CONSOLE(1)] "TypeError: Cannot read property 'getSubscription' of undefined", source: https://www.yahoo.com/service-worker.js (1)
[1129/072307.334:INFO:CONSOLE(1)] "TypeError: Cannot read property 'getSubscription' of undefined", source: https://www.yahoo.com/service-worker.js (1)
[1129/072307.334:INFO:CONSOLE(1)] "TypeError: Cannot read property 'getSubscription' of undefined", source: https://www.yahoo.com/service-worker.js (1)
[1129/072307.335:INFO:CONSOLE(1)] "TypeError: Cannot read property 'getSubscription' of undefined", source: https://www.yahoo.com/service-worker.js (1)
[1129/072308.230:INFO:CONSOLE(3)] "Refused to get unsafe header "Server"", source: https://yep.video.yahoo.com/oath/js/1/oath-player.js?expb=FPDOATH204&ypv=8.2.104 (3)
[1129/072308.994:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1129/072309.377:INFO:CONSOLE(32)] "The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu", source: https://static.adsafeprotected.com/sca.17.4.114.js (32)
[1129/072309.378:INFO:CONSOLE(32)] "a: 0.007080078125 ms", source: https://static.adsafeprotected.com/sca.17.4.114.js (32)
[1129/072355.676:INFO:CONSOLE(0)] "The service worker navigation preload request was cancelled before 'preloadResponse' settled. If you intend to use 'preloadResponse', use waitUntil() or respondWith() to wait for the promise to settle.", source:  (0)
[1129/072355.695:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/072355.996:INFO:CONSOLE(21)] "darla csc writer, potential invalid html content", source: https://s.yimg.com/rq/darla/4-6-0/html/r-csc.html (21)
[1129/072356.026:WARNING:spdy_session.cc(3384)] Received HEADERS for invalid stream 15
[1129/072356.145:INFO:CONSOLE(646)] "A parser-blocking, cross site (i.e. different eTLD+1) script, https://cdn.optimizely.com/js/10685141500.js, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.", source: https://www.intuit.com/intuit/us_sbg_prod/Bootstrap.js (646)
[1129/072356.145:INFO:CONSOLE(646)] "A parser-blocking, cross site (i.e. different eTLD+1) script, https://cdn.optimizely.com/js/10685141500.js, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.", source: https://www.intuit.com/intuit/us_sbg_prod/Bootstrap.js (646)
[1129/072359.330:INFO:CONSOLE(632)] "Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.", source: https://www.intuit.com/intuit/us_sbg_prod/Bootstrap.js (632)
[1129/072417.515:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/072448.793:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1129/072448.879:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1129/072449.067:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1129/072450.064:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1129/072451.201:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1129/072648.906:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1129/072735.071:ERROR:batching_media_log.cc(38)] MediaEvent: {"error":"FFmpegDemuxer: no supported streams"}
[1129/072735.073:ERROR:batching_media_log.cc(35)] MediaEvent: {"pipeline_error":14}
[1129/072752.484:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1129/072804.080:INFO:CONSOLE(523)] "%c%s color: red; background: yellow; font-size: 24px; WARNING!", source: https://www.gstatic.com/_/mss/boq-search/_/js/k=boq-search.VisualFrontendUi.en_US.FzY6j8MJCsY.es5.O/am=lAQECNSXdA8BAAJABAEEiGEIh0GAOEACHBAAgAgACCSCAQAAPgCAAAAAAG54cAgAAAAAABoqAgAAAAAWAQ/d=1/excm=_b,_tp,searchview/ed=1/dg=0/wt=2/ct=zgms/rs=AH7-fg7C-JCfVKGc72BcmA83v_MIM7rFKQ/m=_b,_tp (523)
[1129/072804.080:INFO:CONSOLE(523)] "%c%s font-size: 18px; Using this console may allow attackers to impersonate you and steal your information using an attack called Self-XSS.
Do not enter or paste code that you do not understand.", source: https://www.gstatic.com/_/mss/boq-search/_/js/k=boq-search.VisualFrontendUi.en_US.FzY6j8MJCsY.es5.O/am=lAQECNSXdA8BAAJABAEEiGEIh0GAOEACHBAAgAgACCSCAQAAPgCAAAAAAG54cAgAAAAAABoqAgAAAAAWAQ/d=1/excm=_b,_tp,searchview/ed=1/dg=0/wt=2/ct=zgms/rs=AH7-fg7C-JCfVKGc72BcmA83v_MIM7rFKQ/m=_b,_tp (523)
[1129/072820.103:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
[1129/073448.803:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
[1129/104750.745:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1130/153551.792:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: browser
[1130/153551.911:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: gpu-process
[1130/153552.064:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: utility
[1130/153552.106:INFO:crash_reporting.cc(220)] Crash reporting enabled for process: renderer
[1130/153553.484:INFO:CONSOLE(0)] "Autofocus processing was blocked because a document already has a focused element.", source: https://www.google.com/ (0)
    - Any other background information that's relevant? Are you doing something out of the ordinary? 3rd party controls?
 
- **Does this problem also occur in the `CEF` Sample Application**
 
No - in the zoom I didn't use the args below, but just now I did and still did not see the issue. I also don't see the issue with the WinForms CefSharp sample app.

**
Seems similar to this https://github.com/cefsharp/CefSharp/issues/684 and maybe this https://bitbucket.org/chromiumembedded/cef/issues/2584/text-area-focus-bug-on-all-cef-builds.. although again this is isolated to WPF CefSharp. WinForms is fine.

I'd love advice on a workaround we can do in our code.

Thank you!

**Comment by amaitland** [comment]
Likely a duplicate of #3280

**Comment by ocardoso** [comment]
Thank you! I'm sorry I didn't see that first. We'll try that workaround first thing tomorrow and confirm.

**Comment by ocardoso** [comment]
Confirming.. the work around mentioned in #3280 worked. Thank you!

**Comment by amaitland** [comment]
If the hardware differs to that listed im #3280 then adding additional information to https://github.com/dotnet/wpf/issues/3817 would be helpful. What DPI does the device use?

**Comment by amaitland** [comment]
Can you please confirm the DPI used? The CPU and GPU of the HP envy?

Thanks

**Comment by ocardoso** [comment]
It was this, right out of the box: https://www.bestbuy.com/site/hp-envy-x360-2-in-1-15-6-touch-screen-laptop-intel-core-i7-12gb-memory-512gb-ssd-32gb-optane-natural-silver/6429424.p?skuId=6429424

![image](https://user-images.githubusercontent.com/4204982/101212342-69b79800-362d-11eb-8fda-a3172b32f54f.png)

**Comment by ocardoso** [comment]
Out of the box the "scale and layout" / DPI setting was 125%, but same issues manifested at 100%

**Comment by amaitland** [comment]
> Out of the box the "scale and layout" / DPI setting was 125%, but same issues manifested at 100%

Thanks for confirming. Subscribe to https://github.com/dotnet/wpf/issues/3817 to track the WPF issue.

**Comment by yuhangch** [comment]
Thanks, this title save my day
